### PR TITLE
First steps towards a Progressive Web App

### PR DESCRIPTION
**Pull request by ndorfin**
Hey RubyNZers,

This PR contains:

- heaps of touch-icons in the `/source` folder, so we don't have to declare them in the `<head>`
- an updated `favicon.ico`, contains 16x16 and 32x32 sizes
- centralised name, description and theme settings in `data/site.yml`
- application manifests: `source/manifest.json.erb` and `source/browserconfig.xml`
- better `<head>` usage
- and some smaller misc. additions such as [editorconfig](http://editorconfig.org) <-- Add your editor plugin!

Please shout if you have any questions :)

Next up:
- [x] Measure site performance - see: #30 
- [ ] Preparing for Service Workers: Needs HTTPS first - see:  #28
- [ ] Clean up script dependencies - see: #33 
- [ ] Implement a per-file caching strategy
- [ ] Implement a gzip strategy
- [ ] Implement an asset-hashing strategy
- [ ] Implement Service Workers

### CODE REVIEW

**Comment by mermop** [comment]
Thanks for all this @ndorfin! 
The touch icons are a bit blurry - do you need a bigger version of the icon to export from?

File: data/site.yml [hunk 1]
```diff
@@ -1 +1,6 @@
-name: "Kiwi Ruby"
+name: 'Kiwi Ruby'
+short_name: 'Kiwi Ruby'
+description: 'Kiwi Ruby is a new Ruby conference being held in Wellington, New Zealand in November 2017. It will be very good.'
+theme_color: '#ffffff'
+theme_background_color: '#ffffff'
+url: 'http://kiwi.ruby.nz'
```
Comment (mermop) [inline data/site.yml]: No newline? Also - single quotes?

File: source/partials/_head_breakpoints.html.slim [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
...nction(a,b){-1===a.className.indexOf(b)&&(a.className=""!==a.className?a.className+" "+b:b)},g=function(a,c){var d="breakpoint-"+c,g=b.documentElement;a?f(g,d):e(g,d)},h=function(a){c._globalEvents.forEach(function(b){"function"==typeof b&&b(a)})},i=function(){c._isTicking||j(k),c._isTicking=!0},j=function(b,d){if(!a.requestAnimationFrame){var e=(new Date).getTime(),f=Math.max(0,16-(e-c._debounceLastTime)),g=a.setTimeout(function(){b(e+f)},f);return c._debounceLastTime=e+f,g}a.requestAnimationFrame(b,d)},k=function(){c._isTicking=!1;var b=[];for(var d in c.breakpoints){var e=c.breakpoints[d],f=a.matchMedia(e).matches;if(c._namedEvents[d]&&c._eventMatchCache[d]!==f){c._eventMatchCache[d]=f;for(var i=0;i<c._namedEvents[d].length;i++){var j=c._namedEvents[d][i];"function"==typeof j&&j(f)}}f&&b.push(d),g(f,d)}0!==b.length&&h(b)},l=function(){for(var a=b.getElementsByTagName("meta"),d=0;d<a.length;d++)if("breakpoint"===a[d].name){var e=a[d].getAttribute("content"),f=a[d].getAttribute("media");c.breakpoints[e]=f}},m=function(){l(),k()},n=function(){l(),k(),a.addEventListener("resize",i)};"undefined"!=typeof module&&module.exports?module.exports=c:a.metaQuery=c,m(),d(n)}(window,document);
```
Comment (mermop) [inline source/partials/_head_breakpoints.html.slim]: Again - no newline? I thought standardising that was the point of the editor config..

**Comment by ndorfin** [comment]
Thanks for reviewing @mermop  - I'll make the fixes this week.
I suspect I added EditorConfig after making the partials.

**Comment by ndorfin** [comment]
Hey @mermop - I'm not sure what's up with the last newline, it's showing as a newline in my editor, but not here in this review.

I've reformatted the images: I've been using ImageMagick's `convert`, this time with an unsharp filter. I suspect we might need a different image as the source for the smaller icons, as the stroke-widths are too thin at anything less than 128x128.
Should I use the singular ruby (as used in the favicon) as the source image for smaller touch-icons?

**Comment by mermop** [comment]
The problem with the blurriness might just be that they don't look as good @1x on my retina screen - if I export straight from illustrator it doesn't look that different.

Yes I think it's probably worth using just the ruby for smaller images.

File: source/partials/_head_breakpoints.html.slim (hunk 2 above)
Comment (ndorfin) [reply source/partials/_head_breakpoints.html.slim]: Could this be due to some git config setting? It's bewildering :(

**Comment by ndorfin** [comment]
@mermop The smaller touch icons are now using `favicon.png` as the source image.
I've also updated `manifest.json`.
Good to go when you're ready.

**Review by mermop — APPROVED** [review]
Good to go

File: source/partials/_head_breakpoints.html.slim (hunk 2 above)
Comment (mermop) [reply source/partials/_head_breakpoints.html.slim]: Not coming up with the "no newline" emoji anymore so should be fine 👍

**Comment by ndorfin** [comment]
@mermop I can't merge - you get to push the big green button 😈

**Comment by mermop** [comment]
This one won't have improved the performance as per #30 will it?

**Comment by ndorfin** [comment]
@mermop Nope. I did a test anyway, and it looks a little slower than the previous release, mostly because at this time of the day our pipes are clogged.
See: https://www.webpagetest.org/video/compare.php?tests=170907_VE_736bd99e7d3b840744f76e2b490e7bb2,170906_MX_134b2078407fd429bad17d360910f260,170824_NA_6107a394107fd98c95933811e2478758

We should see some improvement in #33

**Pull request merged**